tests/baselines/reference/user/chrome-devtools-frontend.log

```diff
@@ -3346,6 +3344,10 @@ node_modules/chrome-devtools-frontend/front_end/cm/codemirror.js(5791,33): error
 node_modules/chrome-devtools-frontend/front_end/cm/codemirror.js(5794,12): error TS2339: Property 'atomic' does not exist on type 'TextMarker'.
 node_modules/chrome-devtools-frontend/front_end/cm/codemirror.js(5800,12): error TS2339: Property 'parent' does not exist on type 'TextMarker'.
 node_modules/chrome-devtools-frontend/front_end/cm/codemirror.js(5800,27): error TS2339: Property 'parent' does not exist on type 'TextMarker'.
+node_modules/chrome-devtools-frontend/front_end/cm/codemirror.js(5836,20): error TS2339: Property 'line' does not exist on type 'Pos | { from: Pos; to: Pos; }'.
```

Comment:
It is just on `Pos` however. This is a case where a special declaration has given us a union type for the property, but flow control hasn't rectified the type at this location to just be one or the other. @sandersn weren't you looking at this issue generally recently?

Reply:
In any case, I think it's correct to issue these errors (certainly better than the silent `any`). You can clearly see why we've disallowed the access - it could be one of two things, and we don't know which.

Ah, and this doesn't get a fake access allowed because while the object literal type is in fact a js literal, `Pos` is _not_ it's a function psuedoclass type. AFAIK it actually shouldn't have been allowed before; it's certainly an error if you ask about `Pos` on its own! In TS the `find` method would need to have overloads on the first argument - `1` and `-1` return `Pos`, while any other number returns `{from: Pos, to: Pos}`.

Reply:
Hooo boy, yeah - the idex signature was masking what i'd call a legitimate bug/footgun here. When you had a union like
```ts
{a: number, b: number, [index: string]: any} | {d: string, e: string}
```
we'd allow accesses to `d` and `e` (after all, the index means they might exist in both types, right!?!?) but _not_ `a` and `b`. While that was how we behaved before, I think with special assignments this is a big hole to have left open in JS (using an object literal suddenly makes you lose type info on your classes)! I think it definitely should stay as it is now.